### PR TITLE
chore(htmlhint) feature/add commitizen package, husky and prettier

### PR DESCRIPTION
**Pull request by thedaviddias**
**Fixes**: #255 & #258 

- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

#### Short description of what this resolves:

Add precommit hooks to ensure the uniformity of commit messages and prettier to format the code automatically.

#### Proposed changes:

- Add commitizen and commitlint the same format for commits messages
- Add husky to facilitate the creation of hooks
- Add prettier to format the code automatically

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Shinigami92) [inline .prettierrc.json]: I would prefer two additional settings:
- "arrowParens": "always" (much better readability)
- "printWidth": 100 (works well with github GUI side by side comparison)

My configuration for my own projects looks like this:
```json
{
	"arrowParens": "always",
	"bracketSpacing": true,
	"printWidth": 120,
	"semi": true,
	"singleQuote": true,
	"tabWidth": 4,
	"trailingComma": "none",
	"useTabs": true
}
```

Feel free to select some of these settings

File: package.json [hunk 2]
```diff
@@ -1,74 +1,80 @@
...
-  "bugs": {
-    "url": "https://github.com/thedaviddias/HTMLHint/issues"
-  },
-  "keywords": [
-    "html",
-    "hint",
-    "analysis",
-    "javascript"
-  ],
-  "author": {
```
Comment (Shinigami92) [inline package.json]: I'm missing the "author" and "contributors" :(

File: .prettierrc.json (hunk 1 above)
Comment (thedaviddias) [reply .prettierrc.json]: Yes, we'll be adding more elements in the next patch.

File: package.json (hunk 2 above)
Comment (thedaviddias) [reply package.json]: Don't see why, the list of contributors is given by Github easily and everyone can see it.

Comment (Shinigami92) [reply package.json]: Because you can see the package.json in node_modules 
(It's only for author and main contributors)

**Pull request merged**